### DISCUSSION TITLE
v0.1.5

### DISCUSSION THREAD

**Discussion by Luosuu** [ByteDance-Seed/VeOmni#384] [Announcements]
## Highlights

* Many enhancement on NPU thanks to community contribution, including NPU fused ops and CIs
* Memory-efficient DCP to HuggingFace checkpoint conversion script by @TimYangst 
* Support FlashAttention3 by @Coach257 @Luosuu 

## What's Changed
* [model] fix: quick fix by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/260
* [misc] chore: add auto label github flow & pr title check by @FoolPlayer in  #https://github.com/ByteDance-Seed/VeOmni/pull/264
* [docs] feat: quick start doc for Ascend NPU by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/257
* [dist] fix: the issue where NPU does not support the PreSumMul operation by @Feng0w0 in https://github.com/ByteDance-Seed/VeOmni/pull/263
* [misc] fix: fix auto-label for forked pr by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/277
* [chore] fix: The missing weights_path in build_parallelize_model prevents weight loading when using fsdp2 by @Feng0w0 in https://github.com/ByteDance-Seed/VeOmni/pull/275
* [ci] fix: handle pull_request_target event in auto_label workflow by @Ziyi-Wang in https://github.com/ByteDance-Seed/VeOmni/pull/279
* [misc] fix: fix auto-label logic & add  tag by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/281
* [misc] fix: fix check pr workflow by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/282
* [ops]feat: support npu fused moe by @onehaitao in https://github.com/ByteDance-Seed/VeOmni/pull/269
* [model] fix: calculate norm by foreach by @heidongxianhua in https://github.com/ByteDance-Seed/VeOmni/pull/274
* [feat]Add environment variables that may lead to performance optimization on NPU by @Feng0w0 in https://github.com/ByteDance-Seed/VeOmni/pull/276
* [dist] chore: refactor torch dcp to support multiple storage backends by @Ziyi-Wang in https://github.com/ByteDance-Seed/VeOmni/pull/278
* [config] feat: add qwen3_vl flops estimation by @yfchen-byted in https://github.com/ByteDance-Seed/VeOmni/pull/220
* [config] fix: restrict Python version to 3.11 and fix PyTorch CUDA dependency resolution by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/287
* [dist]fix:fix a bug in the PreMulSum patch by @Feng0w0 in https://github.com/ByteDance-Seed/VeOmni/pull/285
* [ci,config] chore: migrate to uv dependency management with enhanced installation guide by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/232
* [dist] chore: make execute_save public to support multiple DCP backends by @Ziyi-Wang in https://github.com/ByteDance-Seed/VeOmni/pull/288
* [data] fix: video utils by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/294
* [ci] fix: uv by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/292
* [dist] fix: remove redundant code (async_op) in ulysses by @KKZ20 in https://github.com/ByteDance-Seed/VeOmni/pull/295
* [data] fix: interleave iterable dataset sharding by @ValMystletainn in https://github.com/ByteDance-Seed/VeOmni/pull/291
* [misc] fix: print log twice bug by @pjgao in https://github.com/ByteDance-Seed/VeOmni/pull/290
* [data] fix: fix annotation by @Fazziekey in https://github.com/ByteDance-Seed/VeOmni/pull/296
* [docs] chore: update docs with readthedocs by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/286
* [docs] feat: support Readthedocs on npu by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/299
* [misc] fix: del veomni_patch in helper by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/298
* [docs] chore: add icon & fix cite by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/303
* [misc] fix: clean vescale by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/302
* [ops] feat: Optimize Qwen3-Moe Performance on Ascend NPU with Patches by @zhihaofang1017 in https://github.com/ByteDance-Seed/VeOmni/pull/167
* [model, ops] feat: support fa3 by @Luosuu in https://github.com/ByteDance-Seed/VeOmni/pull/305
* [ci] chore: add dockerfile and migrate to run ci with uv cache  by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/300
* [misc] fix: fix mfu for vlm by @heidongxianhua in https://github.com/ByteDance-Seed/VeOmni/pull/301
* [ckpt, data] fix: fix merge dcp & fix chatml by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/310
* [ops] feat: veomni_fa_function by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/304
* [config] refactor: simplify NPU extras with separate aarch64 variant by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/312
* [config] fix: remove torch extra from transformers dependency by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/313
* [ops] fix: mv qwen3moe patch by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/314
* [logging] fix: fix duplicata log print in wandb case by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/315
* [ops] fix: rm ALL_FLASH_ATTENTION_FUNCTIONS by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/316
* [misc] feat: use `MODELING_BACKEND` to control force_hf_load by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/311
* [misc] fix: support enable_full_determinism without enable CUDA_LAUNCH_BLOCKING=1 by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/318
* [model] feat: patch LOSS_MAPPING  by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/321
* [model] feat: fused_moe_forward by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/325
* [docs] feat: add NPU Overview doc and qwen3 training guide doc by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/319
* [config] feat: Set the default value of use_wandb to false by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/324
* [docs] fix: fix npu doc suffix py by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/328
* [ci] test: add e2e ci test by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/320
* [model] feat: support async ulysses for qwen3vl dense by @yyytiancai in https://github.com/ByteDance-Seed/VeOmni/pull/307
* [model] fix: logits when labels None by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/332
* [misc] feat: env param by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/329
* [docs] refactor: Update README by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/334
* [ops] feat: Optimize Qwen3 Performance on Ascend NPU with Patches by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/308
* [data] fix: Add sorted() to os.listdir() result in dataset.py to ensure consistent file order by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/331
* [docs] fix: update readthedoc by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/335
* [ops] feat: Optimize Wan2.1 Performance on Ascend NPU with Patches by @zbian99 in https://github.com/ByteDance-Seed/VeOmni/pull/336
* [dist] fix: loss when sp enabled by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/339
* [misc] fix: vl token count by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/341
* [ckpt] fix: checkpointer kwargs by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/343
* [ops] refactor: abstract commonly used NPU fused operators into ops module by @zhihaofang1017 in https://github.com/ByteDance-Seed/VeOmni/pull/347
* [optim] fix: support fused/foreach adamw on ascend by @onehaitao in https://github.com/ByteDance-Seed/VeOmni/pull/349
* [docs] feat: add wan2.1 training guide doc and mock dataset generator by @zbian99 in https://github.com/ByteDance-Seed/VeOmni/pull/340
* [model] fix: fix config usage in modeling_qwen3_vl by @yfchen-byted in https://github.com/ByteDance-Seed/VeOmni/pull/353
* [misc] fix: bugfix for memory snapshot by @onehaitao in https://github.com/ByteDance-Seed/VeOmni/pull/351
* [ci] fix: add skip remote path test by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/356
* [data] fix: interleave_dataset duplication by @Coach257 in https://github.com/ByteDance-Seed/VeOmni/pull/358
* [ci] fix: change network setting by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/360
* [ci] feat: Add Dockerfile and workflow for Ascend image by @phdddd in https://github.com/ByteDance-Seed/VeOmni/pull/364
* [model] fix: repeat  all2all in Wan2.1 ulysses. by @TKONIY in https://github.com/ByteDance-Seed/VeOmni/pull/333
* [ci] fix: Npu fix by @phdddd in https://github.com/ByteDance-Seed/VeOmni/pull/368
* [ci] fix: fix_npu_proxy by @FoolPlayer in https://github.com/ByteDance-Seed/VeOmni/pull/370
* [ci] test: Npu docker auto build by @phdddd in https://github.com/ByteDance-Seed/VeOmni/pull/372
* [ops] fix: RoPE fused op caused freqs shape bug by @zbian99 in https://github.com/ByteDance-Seed/VeOmni/pull/355
* [ci] fix: sync npu unit tests from gpu unit tests by @onehaitao in https://github.com/ByteDance-Seed/VeOmni/pull/359
* [ckpt, ci] fix: refactor merge_dcp_to_hf.py to remove internal dependencies and add checkpoint verification by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/272
* [ckpt, ci, docs] feat: memory-efficient DCP to HuggingFace checkpoint conversion by @TimYangst in https://github.com/ByteDance-Seed/VeOmni/pull/374
* [ci] fix: fix async ulysses test by @Crystal-jiang in https://github.com/ByteDance-Seed/VeOmni/pull/357
* [ops] feat: add chunkloss characteristics for NPU by @zhihaofang1017 in https://github.com/ByteDance-Seed/VeOmni/pull/361
* [model] feat: Add ascend fused operators for Qwen3VL by @phdddd in https://github.com/ByteDance-Seed/VeOmni/pull/323
* [dist, data] fix: init parallel state in data collator post init to avoid worker processing getting single process state by @Luosuu in https://github.com/ByteDance-Seed/VeOmni/pull/383
* [model, ops] feat: add Qwen3 sequence classification model and loss for embedding classification. by @yiwzhao in https://github.com/ByteDance-Seed/VeOmni/pull/322

## New Contributors
* @yfchen-byted made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/220
* @yyytiancai made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/307
* @TKONIY made their first contribution in https://github.com/ByteDance-Seed/VeOmni/pull/333

**Full Changelog**: https://github.com/ByteDance-Seed/VeOmni/compare/v0.1.4...v0.1.5

<hr /><em>This discussion was created from the release <a href='https://github.com/ByteDance-Seed/VeOmni/releases/tag/v0.1.5'>v0.1.5</a>.</em>
